src/transport/datamodel/utransport.rs

```diff
@@ -11,19 +11,61 @@
  * SPDX-License-Identifier: Apache-2.0
  ********************************************************************************/
 
-use crate::transport::datamodel::{UAttributes, UListener, UPayload, UStatus};
-use crate::uri::datamodel::{UEntity, UUri};
+use async_trait::async_trait;
+use std::sync::mpsc::Sender;
 
+use crate::uprotocol::{UAttributes, UEntity, UMessage, UPayload, UStatus, UUri};
+
+/// `UTransport` is the uP-L1 interface that provides a common API for uE developers to send and receive messages.
+///
+/// Implementations of `UTransport` contain the details for connecting to the underlying transport technology and
+/// sending `UMessage` using the configured technology. For more information, please refer to
+/// [uProtocol Specification](https://github.com/eclipse-uprotocol/uprotocol-spec/blob/main/up-l1/README.adoc).
+#[async_trait]
 pub trait UTransport {
-    /// API to register the calling uE with the underlying transport implementation.
-    fn register(&self, uentity: UEntity, token: &[u8]) -> UStatus;
+    /// Authenticates with the underlying transport layer that the `uEntity` passed
+    /// matches the transport specific identity. This method requires a resolved `UUri`.
+    ///
+    /// # Arguments
+    /// * `uEntity` - Resolved `UEntity`.
+    ///
+    /// # Returns
+    /// Returns `OKSTATUS` if authentication was successful, `FAILSTATUS` if the calling `uE`
+    /// is not authenticated.
+    async fn authenticate(&self, entity: UEntity) -> UStatus;
 
-    /// Transmit UPayload to the topic using the attributes defined in UTransportAttributes.
-    fn send(&self, topic: UUri, payload: UPayload, attributes: UAttributes) -> UStatus;
+    /// Transmits `UPayload` to the topic using the attributes defined in `UTransportAttributes`.
+    ///
+    /// # Arguments
+    /// * `topic` - Resolved `UUri` topic to send the payload to.
+    /// * `payload` - Actual payload.
+    /// * `attributes` - Additional transport attributes.
+    ///
+    /// # Returns
+    /// Returns `OKSTATUS` if the payload has been successfully sent (ACK'ed), otherwise returns
+    /// `FAILSTATUS` with the appropriate failure.
+    async fn send(&self, topic: UUri, payload: UPayload, attributes: UAttributes) -> UStatus;
```

Comment:
The reason why I didn't pick this up yet is that there exists a UStatus::Ok - so what would it mean if I get an Err() from a function that then contains an ::Ok value? ...

Reply:
But UStatus contains a UCode, which can be OK. 
I'm not fundamentally against doing Rust-idiomatic error handling - but I'm not happy about the potential semantics of an Error result that can contain an OK code...

Reply:
A `Result` is an `enum` under the hood, the main difference is that returning a `Result` would allow you to use the `?` to early return in case of errors + all the mapping functionalities which would you need to care about your self. If we want to go for a `Result`, and we want to differentiate different codes for `Ok` and `Err`, the function should then return `Result<UStatusOk, UStatusErr>`. If, there is only one possible case for `Ok`, then `Result<(), UStatus>` seems appropriate.

Reply:
done. But don't ask me what an Err(Ok) would mean... ;-)